fhir-persistence/src/main/java/com/ibm/fhir/persistence/FHIRPersistence.java

```diff
@@ -130,4 +132,13 @@
     default boolean isDeleteSupported() {
         return false;
     }
+
+    /**
+     * Generates a logical identity.
+     * 
+     * @return logical identity
+     */
+    default String getLogicalId() {
+        return UUID.randomUUID().toString();
+    }
```

Comment:
I think because this is in FHIRPersistence this is OK.... but it seems like something we shouldn't have as a default.

Reply:
also the @override declares it as public.  best to mark this as Public in the interface (rather than the default package)

Reply:
I removed the `default` impl and updated the implementing classes to provide a generateResourceId() method. I also explicitly declared all methods in the interface `public`.